### PR TITLE
Add stats Dataframe

### PR DESCRIPTION
**Pull request by DD5HT**
# Description

This code adds a statistics DataFrame function which can be used to analyse ktk datasets containing the following informations:
`File Size, Row Groups, Rows and Partitions`
It's designed to run in parallel on a developer machine.

### CODE REVIEW

File: kartothek/io_components/stats.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+_logger = logging.getLogger(__name__)
+
+
+def get_stats_df(
+    dataset_uuid: str, store_factory: StoreFactory, workers: int = 50
+) -> pd.DataFrame:
+    """
+    Get a statistics DataFrame for a given `dataset_uuid` and `store_factory`
+    This DataFrame contains statistics the following statistics for each `.parquet` file:
+    umber of rows, number of row groups and the file size in bytes.
```
Comment (amerkel2) [inline kartothek/io_components/stats.py]: typo: umber -> number

File: kartothek/io_components/stats.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        A stats DataFrame with the given columns:
+        ```
+        ["key", "num_rows", "num_row_group", "file_size", "partition]
+        ```
+    """
+    ds_factory = DatasetFactory(dataset_uuid, store_factory)
+    ds_keys = list(_get_dataset_keys(ds_factory))
+    get_data = partial(_fetch_stats, store_factory)
+
+    with ThreadPoolExecutor(max_workers=workers) as executer:
```
Comment (lr4d) [inline kartothek/io_components/stats.py]: It's `executor`

**Comment by lr4d** [comment]
I think this should be in a module such as `kartothek.util` as `io_components` is used to define read/write/update/index/gc functions and this arguably falls out of this scope

File: kartothek/io_components/stats.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    Returns
+    -------
+    pd.DataFrame
+        A stats DataFrame with the given columns:
+        ```
+        ["key", "num_rows", "num_row_group", "file_size", "partition]
+        ```
+    """
+    ds_factory = DatasetFactory(dataset_uuid, store_factory)
+    ds_keys = list(_get_dataset_keys(ds_factory))
```
Comment (fjetter) [inline kartothek/io_components/stats.py]: Why implement your own get_dataset_keys function? From my understanding it parses the DatasetMetadata and returns *all* keys, i.e. also `_common_metadata` and the `by-dataset-metadata.json` although you're filtering in `get_data` again for `.parquet` files. If you're only interested in the parquet files you can also use the `dispatch_metapartitions` functionality.

I would even suggest to implement a `MetaPartition.get_statistics` method. This way you could essentially run 

```
mp_generator = dispatch_metapartitions_from_factory(...)
with ThreadPoolExecutor(max_workers=workers) as executor:
     data = executor.map(MetaPartition.get_statistics, mp_generator)
```

**Comment by lr4d** [comment]
@DD5HT what's the status of this?

**Comment by fjetter** [comment]
closing as stale. @DD5HT if you want to pick this up again, feel free to reopen

**Pull request closed without merging**